pkg/devserver/api.go

```diff
@@ -178,6 +211,30 @@ func (a devapi) Register(w http.ResponseWriter, r *http.Request) {
 	_, _ = w.Write([]byte(`{"ok":true}`))
 }
 
+// hashSigningKey hashes the signing key for comparison with authorization
+// headers in requests from an SDK.
+//
+// It accounts for keys that are not strictly hex strings as users could specify
+// a key of any length.
+func hashSigningKey(signingKey *string) string {
+	if signingKey == nil || *signingKey == "" {
+		return ""
+	}
+
+	// Because we expect the signing key to be a hex string due to being this
+	// format in Cloud, it needs to have an even number of characters. We pad
+	// here with a leading 0 if it doesn't, as the client SDKs should be when
+	// sending the key.
+	encodedKey := *signingKey
+	if len(encodedKey)%2 != 0 {
+		encodedKey = "0" + encodedKey
+	}
```

Comment:
cc @goodoldneon This is a quirk of the TS SDK's hashing to ensure it's correctly hashing a hex key, but it may not be present everywhere. 🤔

Reply:
Enforcing that keys must be hex for now.